addon/components/ember-tutorial-component.js

```diff
@@ -191,27 +223,120 @@ export default Ember.Component.extend({
       this.set('hideMessage', false);
       this.set('currentConfig', this.get('config').data[0]);
       run.schedule('afterRender', this, function() {
-        this.initTootltip();
+        this.initTooltip();
       });
     } else {
       this.set('hideMessage', true);
     }
   },
 
   /**
+   * @method turn off resize listeners
+   */
+  cancelAutoResize() {
+    console.log('cancelAutoResize');
+    Ember.$(window).off('resize.tutorial-component');
+  },
+
+  /**
    * @method  updateConfigDetails
    */
   updateConfigDetails(nextConfigIndex) {
     this.set('currentConfig', this.get('config').data[nextConfigIndex]);
     this.set('currentConfigIndex', nextConfigIndex);
     run.scheduleOnce('afterRender', this, function() {
-      this.computeXCord();
-      this.computeYCord();
+      this.setPosition();
       window.scrollTo(this.get('xCoord'), this.get('yCoord'));
     });
   },
 
   /**
+   * Dynamically calculate the border width of given element
+   * @param  {String} element
+   * @param  {String} pseudoElement
+   * @return {Number}                 the border width in px
+   */
+  getBorderWidth(elementSelector='.tutorial-component', pseudoElementSelector=':before') {
+    const properties = ['border-top-width', 'border-left-width', 'border-right-width', 'border-bottom-width'];
+    const borderWidths = properties.map((property) => {
+        const borderWidth = window.getComputedStyle(document.querySelector('.tutorial-component'), pseudoElementSelector).getPropertyValue(property);
+        return parseInt(borderWidth);
+    });
+
+    const maxBorderWidth = Math.max.apply(null, borderWidths);
+    return maxBorderWidth;
+  },
+
+  /**
+  * Either call
+  * 1) computeXYCord, or
+  * 2) computeXCord and computeYCord separately
+  * Based on the given pointer direction.
+  * If pointerDirection's given, call computerXYCord
+  * otherwise call computeXCord and computeYCord
+  */
+  setPosition() {
+    if (this.get('currentConfig').pointerDirection) {
+      this.computeXYCord();
+    } else {
+      this.computeXCord();
+      this.computeYCord();
+    }
+  },
+
+  computeXYCord(tooltipElementSelector='.tutorial-component') {
```

Comment:
Please add some documentation every where. Also if possible add some inline comments explaining the code.